common/test/acceptance/pages/studio/course_info.py

```diff
@@ -1,16 +1,169 @@
+        """
+        return self.q(css='.handouts-content').present
+
+    def are_course_updates_on_page(self):
+        """
+        Is the course updates content Div loaded on the page
+        """
+        return self.q(css='article#course-update-view.course-updates').present
+
+    def is_new_update_button_present(self):
+        """
+        Checks for the presence of the new update post button
+        """
+        return self.q(css='.new-update-button').present
+
+    def click_new_update_button(self):
+        """
+        Wait for the handouts-content, otherwise the render of the page will clear the new update
+        form when loading content
+        """
+        self.wait_for_element_presence('.handouts-content', '.handouts-content')
+        self.q(css='.new-update-button').click()
+        self.wait_for_page()
+        self.wait_for_element_visibility('.CodeMirror', 'Waiting for .CodeMirror')
+
+    def is_new_update_form_present(self):
+        """
+        Has the new update form loaded and it is present
+        """
+        return self.q(css='.CodeMirror').present
+
+    def submit_update(self, message):
+        """
+        Adds update text to the new update CodeMirror form and submits that text
+
+        Arguments:
+            message (str): The message to be added and saved
+        """
+        type_in_codemirror(self, 0, message)
+        if self.is_new_update_save_button_present():
+            self.click_new_update_save_button()
+            self.wait_for_page()
+
+    def set_date(self, date):
+        """
+        Sets the updates date input to the provided value
+
+        Arguments:
+            date (str): Date string in the format DD/MM/YYYY
+        """
+        set_input_value(self, 'input.date', date)
+
+    def is_update_date(self, search_date):
+        """
+        Checks to see if the search date is present
+
+        Arguments:
+            search_date (str): e.g. 06/01/2013 would be found with June 1, 2013
+        """
+        dates = self.q(css='.date-display').html
+        for date in dates:
+            if search_date == date:
+                return True
+        return False
+
+    def is_new_update_save_button_present(self):
+        """
+        Checks to see if the CodeMirror Update save button is present
+        """
+        return self.q(css='.save-button').present
+
+    def click_new_update_save_button(self):
+        """
+        Clicks the CodeMirror Update save button
+        """
+        self.q(css='.save-button').first.click()
+        self.wait_for_page()
+
+    def is_edit_button_present(self):
+        """
+        Checks to see if the edit update post buttons if present
+        """
+        return self.q(css='div.post-preview .edit-button').present
+
+    def click_edit_update_button(self):
+        """
+        Clicks the edit update post button
+        """
+        self.q(css='div.post-preview .edit-button').first.click()
+        self.wait_for_page()
```

Comment:
No need for waite_for_page. You might want to wait though until the CodeMirror edit window is ready.